publish.js

```diff
@@ -0,0 +1,188 @@
+            fs.removeSync(path.join(SCENARIOS, course.name));
+        }
+    });
+
+    SPECIFIED_COURSES.forEach(course => {
+        if(course.name.includes('-pathway')){
+            fs.copyFile(path.join(PLAYBOOKS, course.name), path.join(SCENARIOS, course.name));
+            console.log(course.name, 'updated.' );
+        }
+    });
+}
+
+function generateNewScenarios(){
+    let cp = child_process.spawnSync(`bash buildRun.sh -e katacoda`, { shell: true, cwd: COMPILER, encoding: 'utf-8' });
+    if(!fs.existsSync(path.join(COMPILER, 'build', 'output', 'katacoda'))) {
+        console.log("ERROR[generateNewScenarios]: ", cp);
+        return "";
+    }
+}
+
+function createCourse(coursefile){
+    //create course folder
+    let courseDirName = coursefile.replace("-pathway.json", '');
+    createFolder(path.join(TEMP_COURSES, courseDirName));
+    folderNames.push(courseDirName);
+
+    //parse scenario names
+    const genScenariosDir = path.join(COMPILER, 'build', 'output', 'katacoda');
+    let rawData = fs.readFileSync(path.join(SCENARIOS, coursefile));
+    let courseJson = JSON.parse(rawData);
+
+    courseJson.courses.forEach(scenario => {
+        fs.copySync(path.join(genScenariosDir, scenario.course_id), path.join(TEMP_COURSES, courseDirName, scenario.course_id));
+        if(!fs.existsSync(path.join(TEMP_COURSES, courseDirName, scenario.course_id))) {
+            console.log('Copying', scenario.course_id, 'to', courseDirName, 'failed');
+            return "";
+        }
+        if(!usedScenarios.includes(scenario.course_id)){
+            usedScenarios.push(scenario.course_id);
+        }
+    });
+
+    console.log(courseDirName, 'created in temporory directory.' );
+    
+}
+
+function createScenario(scenario){
+    const genScenariosDir = path.join(COMPILER, 'build', 'output', 'katacoda');
+    folderNames.push(scenario);
+    fs.copySync(path.join(genScenariosDir, scenario), path.join(TEMP_FILES, scenario));
+
+    console.log(scenario, 'created in temporory directory.' );
+}
+
+function cleanUp(){
+    const ONLINE_COURSES = getCourses(SCENARIOS); 
+    const ONLINE_FOLDERS = getScenarios(SCENARIOS);
+
+    ONLINE_COURSES.forEach(coursefile => {
+        course = coursefile.replace('-pathway.json');
+        if(! folderNames.includes(course)){
+            fs.removeSync(path.join(SCENARIOS, course) ,{ recursive: true });
+            console.log(course, "doesn't exist anymore in the tutorials repository." );
+        }
+    });
+
+    ONLINE_FOLDERS.forEach(scenario => {
+        if(! folderNames.includes(scenario)){
+            fs.removeSync(path.join(SCENARIOS, scenario) ,{ recursive: true });
+            console.log(scenario, "doesn't exist anymore in the tutorials repository." );
+        }
+    });
+
+    //TODO Filter only newly generated folders
+    //ISSUE #25 #26
+    ONLINE_FOLDERS.forEach(scenario => {
+        fs.removeSync(path.join(SCENARIOS, scenario) ,{ recursive: true });
```

Comment:
Here It might delete some folder, that is needed in the kataocoda-scenario but not listed in the tutorial repository. 
I am not sure how to differentiate in this case between normal folders and tutorials